processing/src/main/java/io/druid/segment/ColumnSelectorFactory.java

```diff
@@ -30,4 +31,5 @@
   public FloatColumnSelector makeFloatColumnSelector(String columnName);
   public LongColumnSelector makeLongColumnSelector(String columnName);
   public ObjectColumnSelector makeObjectColumnSelector(String columnName);
+  public ValueType columnType(String columnName);
```

Comment:
looking at this code more, I don't really like the way that the column selector factory, which is responsible for creating column selectors, has a get type method